packages/types/src/transport.ts

```diff
@@ -13,6 +15,47 @@ export type Outcome =
   | 'ratelimit_backoff'
   | 'sample_rate';
 
+export type TransportCategory = 'error' | 'transaction' | 'attachment' | 'session';
+
+export type TransportRequest = {
+  body: string;
+  category: TransportCategory;
+};
+
+export type TransportMakeRequestResponse = {
+  body?: string;
+  headers?: {
+    [key: string]: string | null;
+    'x-sentry-rate-limits': string | null;
+    'retry-after': string | null;
+  };
+  reason?: string;
+  statusCode: number;
+};
+
+export type TransportResponse = {
+  status: EventStatus;
+  reason?: string;
+};
+
+export interface InternalBaseTransportOptions {
+  bufferSize?: number;
+}
+export interface BaseTransportOptions extends InternalBaseTransportOptions {
+  // url to send the event
+  // transport does not care about dsn specific - client should take care of
+  // parsing and figuring that out
+  // If no DSN is configured, pass in an undefined url.
+  url: string;
```

Comment:
If users are supposed to pass in undefined, should we type it as such?

Reply:
The comment is wrong 😓 - will fix